client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -54,42 +56,94 @@ const RequestGroupForm: FunctionComponent<Props> = (props: Props) => {
     "https://source.unsplash.com/FOrNjVWdUzM",
   ]
 
+  const createRequestGroupMutation = gql`
+  mutation CreateRequestGroup(
+    $name: String!
+    $description: String!
+    $image: String!
+    $requestTypeNames: [String!]!
+  ) {
+    createRequestGroup(requestGroup: {
+      name: $name
+      description: $description
+      image: $image
+      requestTypeNames: $requestTypeNames
+    }) {
+      success
+      message
+      id
+    }
+  }`
+
+  const updateRequestGroupMutation = gql`
+  mutation UpdateRequestGroup(
+    $id: ID!, 
+    $name: String!, 
+    $description: String!,
+    $image: String!
+    $requestTypeNames: [String!]!
+  ) {
+    updateRequestGroup(requestGroup: {
+      id: $id
+      name: $name
+      description: $description
+      image: $image
+      requestTypeNames: $requestTypeNames
+    }) {
+      success
+      message
+      id
+    }
+  }`
+
+  const [createRequestGroup] = useMutation(createRequestGroupMutation);
+  const [updateRequestGroup] = useMutation(updateRequestGroupMutation);
+
 
   const requestGroupQuery = gql`
   query FetchRequestGroup($id: ID!) {
     requestGroup(id: $id) {
-      id
+      _id
       name
       description
       image
       requestTypes {
-        id
+        _id
         name
         deleted
       }
     }
   }`
 
-  if (props.requestGroupId) {
-    useQuery(requestGroupQuery, {
+  if (props.operation === "edit") {
+    const { loading } = useQuery(requestGroupQuery, {
       variables: { id: props.requestGroupId, },
+      fetchPolicy: 'network-only',
       onCompleted: (data: { requestGroup: RequestGroup }) => {
-        const res: RequestGroup = JSON.parse(JSON.stringify(data.requestGroup)); // deep-copy since data object is frozen
-
-        setInitialRequestGroup(res)
-        setName(res.name ? res.name : "")
-        setDescription(res.description ? res.description : "")
-        setImage(res.image ? res.image : "")
+        const retrievedRequestGroup: RequestGroup = JSON.parse(JSON.stringify(data.requestGroup)); // deep-copy since data object is frozen
+        
+        setInitialRequestGroup(retrievedRequestGroup)
+        setName(retrievedRequestGroup.name ? retrievedRequestGroup.name : "")
+        setDescription(retrievedRequestGroup.description ? retrievedRequestGroup.description : "")
+        setImage(retrievedRequestGroup.image ? retrievedRequestGroup.image : "")
 
         setRequestTypeNames(
-          res.requestTypes ?
-            res.requestTypes
-              .filter((requestType) => requestType.name)
+          retrievedRequestGroup.requestTypes ?
+            retrievedRequestGroup.requestTypes
```

Comment:
I saw that you added requestTypeNames to the schema in the server so I was wondering if that could be used here? Or maybe it's not necessary to add to the schema